errors.go

```diff
@@ -11,41 +11,46 @@ import (
 type (
 	// Errors may be generated during deserialization, binding,
 	// or validation. It implements the built-in error interface.
-	Errors []Error
+	Errors []FieldError
 
 	// Error is a powerful implementation of the built-in error
 	// interface that allows for error classification, custom error
 	// messages associated with specific fields, or with no
 	// associations at all.
-	Error struct {
-		// An error supports zero or more field names, because an
-		// error can morph three ways: (1) it can indicate something
-		// wrong with the request as a whole, (2) it can point to a
-		// specific problem with a particular input field, or (3) it
-		// can span multiple related input fields.
-		FieldNames []string `json:"fieldNames,omitempty"`
+	fieldError struct {
+		// An error supports a field names because an
+		// error can morph two ways: (1) it can indicate something
+		// wrong with the request as a whole, or (2) it can point to a
+		// specific problem with a particular input field.
+		fieldName string `json:"fieldNames,omitempty"`
```

Comment:
You're right on all counts. :+1: A user's custom Validate method could, however, make an error that spanned 0 fields or 2+ fields (which we did at my last company). I'll take a look at the rest of this after the holiday (state holiday, go figure!) - thanks for working on this!